pandas/io/excel/_base.py

```diff
@@ -1064,21 +1072,18 @@ def __init__(
             )
 
         if engine is None:
-            if ext == "ods":
-                engine = "odf"
-            elif ext == "xls":
-                engine = "xlrd"
-            else:
-                # GH 35029 - Prefer openpyxl except for xls files
-                if (
-                    import_optional_dependency(
-                        "openpyxl", raise_on_missing=False, on_version="ignore"
-                    )
-                    is not None
-                ):
-                    engine = "openpyxl"
-                else:
-                    engine = "xlrd"
+            if (
+                import_optional_dependency(
+                    "openpyxl", raise_on_missing=False, on_version="warn"
+                )
+                is None
+                and xlrd_version is not None
+            ):
+                config.set_option("io.excel.xlsx.reader", "xlrd")
```

Comment:
hmm this is mutating a user option. can you rather just set engine here?

Reply:
I think this should be moved to `get_default_engine`. It should just adjust the return value, not modify the user option.